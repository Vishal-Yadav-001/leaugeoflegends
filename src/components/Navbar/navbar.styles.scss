@import '../../styles/variables.styles.scss';
.MuiToolbar-regular {
  background-color: $secondary;
  height: 4rem;
}
.list_container {
  width: 100%;
  display: flex;
}
.MuiListItem-root.list_item_1 {
  width: 50%;
  justify-content: start;
}
.MuiListItem-root.list_item_2 {
  width: 50%;
  justify-content: end;
}
.MuiListItem-root.list_item_1 h3 {
  font-size: clamp(1rem, 2.5vw, 1.8rem);
  color: $neutral;
}
.logo {
  width: clamp(1.2rem, 2.5vw, 1.8rem);
}
.badge .MuiBadge-badge{
  background-color: $accent;
  color: $black;
}