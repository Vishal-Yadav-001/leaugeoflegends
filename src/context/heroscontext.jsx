import { createContext, useState } from "react";

const HerosContext = createContext();

const HerosProvider = ({ children }) => {
  const [favoriteHero, setFavoriteHero] = useState([]);
  const [isHeroPresent, setIsHeroPresent] = useState([]);
  /**
   * Add selected hero users watch list
   * @param {*} - value
   */
  const addHero = (value) => {
    setFavoriteHero((prev) => {
      if (prev && prev.includes(value)) {
        return [...prev];
      } else {
        setIsHeroPresent((prev)=> [...prev,value?.id])
        return [...prev, value];
      }
    });
  };
  /**
   * Check if hero is already present in user's watchlist
   * @param {number} id - id of the selected hero
   * @returns {boolean} true if hero already exist else false
   */
  const checkIsHeroPresent = (id) => {
    return isHeroPresent?.includes(id) ? true : false;
  };
  /**
   * Removes selected hero from user's watchlist
   * @param {number} id 
   */
  const handleRemoveRecord = (id)=>{
    setFavoriteHero((prev)=> prev.filter(data=> data.id !== id));
    setIsHeroPresent((prev)=> prev.filter(data=> data !== id));
   }
  return (
    <HerosContext.Provider
      value={{
        favoriteHero,
        setFavoriteHero,
        addHero,
        checkIsHeroPresent,
        isHeroPresent,
        setIsHeroPresent,
        handleRemoveRecord
      }}
    >
      {children}
    </HerosContext.Provider>
  );
};

export { HerosContext, HerosProvider };
