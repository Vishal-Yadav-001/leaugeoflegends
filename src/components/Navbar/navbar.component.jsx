import {
  AppBar,
  Badge,
  IconButton,
  List,
  ListItem,
  Menu,
  Toolbar,
  Tooltip,
  Zoom,
} from "@mui/material";
import React, { Fragment, useContext } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import { HerosContext } from "../../context/heroscontext";
import "./navbar.styles.scss";
const Navbar = () => {
    /** to navigate between routes  */
  const navigate = useNavigate();
    /** fetch router location object keys {pathname, search}  */
  const location = useLocation();
  const {favoriteHero} = useContext(HerosContext);
  return (
    <Fragment>
      <AppBar>
        <Toolbar>
          <List className="list_container">
            <ListItem className="list_item_1">
            <Tooltip TransitionComponent={Zoom} title="Home" placeholder="bottom">
              <IconButton className="home" onClick={()=> { location?.pathname !=="/" && navigate("/")}}>
                <img
                  className="logo"
                  alt="logo"
                  src={require("../../assets/swords.png")}
                ></img>
              </IconButton>
              </Tooltip>
              <h3>Game Store</h3>
            </ListItem>
            <ListItem className="list_item_2">
              <Tooltip TransitionComponent={Zoom} title="Watchlist" placeholder="bottom">
              <IconButton className="watchlist" onClick={()=> { location?.pathname !=="/favorite" && navigate("/favorite")}} >
                <Badge className="badge" badgeContent={favoriteHero?.length} color="primary">

                <img
                  alt="watch list"
                  className="logo"
                  src={require("../../assets/bookmark.png")}
                  ></img>
                  </Badge>
              </IconButton>
              </Tooltip>
            </ListItem>
          </List>
        </Toolbar>
      </AppBar>
    </Fragment>
  );
};

export default Navbar;
