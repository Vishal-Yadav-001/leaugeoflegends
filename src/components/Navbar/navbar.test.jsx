import { act, fireEvent, render, screen, within } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { HerosProvider } from "../../context/heroscontext";
import HerosList from "../Heros List/herosList.component";
import WatchList from "../Watchlist/watchlist.component";
import Navbar from "./navbar.component";
import renderer from "react-test-renderer";
describe("<Navbar/>", () => {
  test("snapshot", () => {
    const navbar = renderer
      .create(
        <BrowserRouter>
          <HerosProvider>
            <Navbar />
          </HerosProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
  test("should render the navigation bar and display app name", () => {
    const navbar = renderNavbar();
    const title = navbar.getAllByText("Game Store");
    const titleText = within(title[0]).getByText("Game Store");
    expect(titleText).toHaveTextContent("Game Store");
  });
  test("should render the navigation bar and test home button", () => {
    const navbar = renderNavbar();
    const homeIcon = screen.queryAllByRole("button")[0];
    act(() => {
      fireEvent.click(homeIcon);
      const layout = render(
        <HerosProvider>
          <HerosList />
        </HerosProvider>,
        { wrapper: BrowserRouter }
      );
      expect(layout).toBeTruthy();
    });
  });
  test("should render the navigation bar and test watchlist button", () => {
    const navbar = renderNavbar();
    const watchlistIcon = screen.queryAllByRole("button")[1];
    act(() => {
      fireEvent.click(watchlistIcon);
      const watchlist = render(
        <HerosProvider>
          <WatchList />
        </HerosProvider>,
        { wrapper: BrowserRouter }
      );
      expect(watchlist).toBeTruthy();
    });
  });
});

const renderNavbar = () => {
  return (
    render(
      <BrowserRouter>
        <HerosProvider>
          <Navbar />
        </HerosProvider>
      </BrowserRouter>
    )
  )
}