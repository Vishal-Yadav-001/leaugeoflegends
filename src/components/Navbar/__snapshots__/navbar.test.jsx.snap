// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <ul
      className="MuiList-root MuiList-padding list_container css-h4y409-MuiList-root"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding list_item_1 css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <button
          aria-label="Home"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium home css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          placeholder="bottom"
          tabIndex={0}
          type="button"
        >
          <img
            alt="logo"
            className="logo"
            src="swords.png"
          />
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <h3>
          Game Store
        </h3>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding list_item_2 css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <button
          aria-label="Watchlist"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium watchlist css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          placeholder="bottom"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiBadge-root badge BaseBadge-root css-1c32n2y-MuiBadge-root"
          >
            <img
              alt="watch list"
              className="logo"
              src="bookmark.png"
            />
            <span
              className="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary BaseBadge-badge BaseBadge-invisible css-1pi4uwz-MuiBadge-badge"
            />
          </span>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</header>
`;
